src/Illuminate/Database/Concerns/BuildsQueries.php

```diff
@@ -158,4 +158,15 @@ protected function simplePaginator($items, $perPage, $currentPage, $options)
             'items', 'perPage', 'currentPage', 'options'
         ));
     }
+
+    /**
+     * Determine if the object is a acceptable callable for query builder.
+     *
+     * @param mixed $callable
+     * @return bool
+     */
+    protected function isValidCallable($callable)
+    {
+        return $callable && ! is_string($callable) && ! is_array($callable) && is_callable($callable);
```

Comment:
Isn't `[$this, 'method']` invokable / callable, but also array?

Reply:
`! is_array($callable)` prevents that

Reply:
the problem is here that a global object might exist, it's not necessary for it to be `$this` as a first arg, it could be `['foo', 'bar']` and it's still a valid callable if foo exists

Reply:
`['foo', 'bar']` is actually the main reason why a validation test failed the first time I made the pull request, one of the tests created it (didn't really dig into which one), and later on it was recognized as a valid callable. If you run validation test separately, everything works fine.

Reply:
That's my point, you should not do `! is_array()` as it's a valid callable. Why should it be allowed?

Reply:
I cannot do that for safety, specifically the `whereIn` queries. `whereIn` query allows `$value` to be an array. That means if you do `$query->whereIn('type', ['Notification', 'Error'])`, it will be recognized as a valid callable instead of just a list of types, because a `Notification` facade exists.